### DISCUSSION TITLE
TypeError: Object(...) is not a function at createPinia

### DISCUSSION THREAD

**Discussion by leopons** [vuejs/pinia#1148] [Help and Questions] [answered]
Hi! I can't seem to run Pinia at all with Vue3, I get a browser console error at createPinia.

```
Uncaught TypeError: Object(...) is not a function
    at createPinia (pinia.mjs?be92:907:1)
```

Here's my `app.js`
```
import { createApp } from "vue";
import App from "./App.vue";
import { createPinia } from "pinia";

createApp(App).use(createPinia()).mount("#app");
```
(actually this is not my real `app.js` but I removed everything else to keep it as simple as possible here)

And my `dependencies`:
```
    "core-js": "^3.6.5",
    "pinia": "^2.0.11",
    "vue": "^3.0.0",
    "vue-router": "^4.0.0-0",
    "vue-select": "^4.0.0-beta.3",
    "vue-svg-inline-plugin": "^2.2.0"
```

I am using Vue CLI, and I also tried with the unofficial plugin as described in the [docs](https://pinia.vuejs.org/getting-started.html) but without success (do I have to do something more than just yarn add it ?)

Also might be worth mentioning I had [this problem](https://github.com/vuejs/pinia/issues/675) and fixed it as described [here](https://github.com/vuejs/pinia/issues/675#issuecomment-919544784), even though I have to say I may not grasp completely what's happening here.

I have to say I found nothing about this and don't know where to search anymore. Maybe I'm not seeing something obvious, but I'll really appreciate some help

**Comment by dodocie** [comment]
got same error and found no solutions yet

**Reply by dodocie** [reply, marked as answer]
fixed this error by upgrade VUE3 bcoz difference version of @vue/reactivity --- version @3.1.1 to @3.2.31.
there's no 'effectScope' exported in dependency @vue/reactivity @3.1.1

**Reply by leopons** [reply]
Thanks! Works for me too. I guess this should be ensured in pinia dependencies

**Reply by mirko77** [reply]
Same issue here

**Comment by lukaszkups** [comment]
I'm working with Vue < 3 and have the same problem/error - my `vue.config.js` file has the webpack rule mentioned in the answers for the questions mentioned by the author already, but that doesn't fix the issue - any recommendations?

**Reply by weiaohan** [reply]
I'm working with vue@2.6.14 and I'm fixed this error by downgrade nodejs from 16 to14

**Reply by 60late** [reply]
I'm using vue@2.6 and pnpm@9.15.
In case anyone who need it， I solved this porblem by using `pnpm approve-builds` to allow all postinstall action. Otherwise , pacakges like `corejs` were not installed correctly。
So I guess this is not a problem with pinia, it's a problem that some of your package were not right.Try to install again may solve this problem.

**Comment by neuhxy90** [comment]
I update vue from @3.1.5 to **@3.2.45** and it works for me.

`yarn add vue@^3.2.45`

`"vue": "^3.2.45",`

**Comment by Timakra** [comment]
just in case some one has this issue notice that pinia version 2.1.0 and update requires  Vue 3.3 - so this might lead to the same issue

**Comment by SendorWang** [comment]
Now I got same error  with vue 3.2.47:

```
    "pinia": "^2.1.4",
    "vue": "^3.2.47",
    "vue-router": "^4.2.2"
```

**Reply by Timakra** [reply]
Try down grading pinia to below 2.1.0 as I mentioned for version above 2.1.0 you need vue 3.3.x

**Reply by SendorWang** [reply]
I have a mistake. Very stupid.
Destructured state and getters:  const {statename,  gettername, ... } = storeToRefs(store)
Destructured action：
  const { state1, getter1 } = storeToRefs(store) //×
  const { action1, action2 } = store //√

**Reply by Timakra** [reply]
The way you did it will make the state lose reactivity it should be 
```
const {actionname, ... } = storeToRefs(store())
```
Notice invocing the `store`

**Reply by SendorWang** [reply]
Thanks.
